src/implementation/Client/GRPCClient/state.ts

```diff
@@ -45,7 +46,11 @@ export default class GRPCClientState implements IClientState {
     this.client = client;
   }
 
-  async save(storeName: string, stateObjects: KeyValuePairType[]): Promise<void> {
+  async save(
+    storeName: string,
+    stateObjects: KeyValuePairType[],
+    metadata?: KeyValueType,
```

Comment:
Shouldn't we make this an option parameter?

Reply:
We can do that, good idea. Updated.